### PR TITLE
Module #6 - Advanced Multithreading

### PR DESCRIPTION
**Pull request by OlenaStoliarova**

### CODE REVIEW

File: m6-advanced-multithreading/t3-file-scanner-fjp/src/main/java/pl/mentoring/filescanner/DirStatisticsProvider.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        ForkJoinPool pool = new ForkJoinPool(processorsCount);
+
+        try {
+            return pool.invoke(new DirStatisticsWorker(dir));
+        } finally {
+            pool.shutdown();
+        }
+    }
+
+    private class DirStatisticsWorker extends RecursiveTask<DirStatistics> {
```
Comment (swepsa) [inline m6-advanced-multithreading/t3-file-scanner-fjp/src/main/java/pl/mentoring/filescanner/DirStatisticsProvider.java:33]: Can you explain why it is not static class?

File: m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    void compute() {
+        assertEquals(1134903170L, new ForkJoinPool().invoke(new FibonacciTask(45)).longValue());
+    }
+
+    @Test
+    void compute_shouldReturnExpectedValue() {
+        ForkJoinPool pool = new ForkJoinPool();
+        assertEquals(1L, pool.invoke(new FibonacciTask(2)).longValue());
+        assertEquals(8L, pool.invoke(new FibonacciTask(6)).longValue());
+        assertEquals(6765L, pool.invoke(new FibonacciTask(20)).longValue());
```
Comment (swepsa) [inline m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java:21]: Unnecessary unboxing

File: m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.concurrent.ForkJoinPool;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FibonacciTaskTest {
+
+    @Test
+    void compute() {
+        assertEquals(1134903170L, new ForkJoinPool().invoke(new FibonacciTask(45)).longValue());
```
Comment (swepsa) [inline m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java:13]: Unnecessary unboxing

File: m6-advanced-multithreading/t7-forkjoin/src/main/java/pl/mentoring/forkjoin/ForkBlur.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        int sidePixels = (mBlurWidth - 1) / 2;
+        for (int index = mStart; index < mStart + mLength; index++) {
+            // Calculate average.
+            float rt = 0, gt = 0, bt = 0;
+            for (int mi = -sidePixels; mi <= sidePixels; mi++) {
+                int mindex = Math.min(Math.max(mi + index, 0), mSource.length - 1);
+                int pixel = mSource[mindex];
+                rt += (float) ((pixel & 0x00ff0000) >> 16) / mBlurWidth;
+                gt += (float) ((pixel & 0x0000ff00) >> 8) / mBlurWidth;
+                bt += (float) ((pixel & 0x000000ff) >> 0) / mBlurWidth;
```
Comment (swepsa) [inline m6-advanced-multithreading/t7-forkjoin/src/main/java/pl/mentoring/forkjoin/ForkBlur.java:40]: What does (pixel & 0x000000ff) >> 0 do?

**Review by swepsa — APPROVED** [review]

File: m6-advanced-multithreading/t3-file-scanner-fjp/src/main/java/pl/mentoring/filescanner/DirStatisticsProvider.java (hunk 1 above)
Comment (OlenaStoliarova) [reply m6-advanced-multithreading/t3-file-scanner-fjp/src/main/java/pl/mentoring/filescanner/DirStatisticsProvider.java:33]: There is no good reason for that. Changed to static

File: m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java (hunk 2 above)
Comment (OlenaStoliarova) [reply m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java:21]: fixed

File: m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java (hunk 3 above)
Comment (OlenaStoliarova) [reply m6-advanced-multithreading/t6-recursivetask-recursiveaction/src/test/java/pl/mentoring/recursive/FibonacciTaskTest.java:13]: fixed

File: m6-advanced-multithreading/t7-forkjoin/src/main/java/pl/mentoring/forkjoin/ForkBlur.java (hunk 4 above)
Comment (OlenaStoliarova) [reply m6-advanced-multithreading/t7-forkjoin/src/main/java/pl/mentoring/forkjoin/ForkBlur.java:40]: My guess - makes formulas uniform and easier to read.
It's code from tutorial, I used it as is: https://docs.oracle.com/javase/tutorial/essential/concurrency/forkjoin.html

**Pull request merged**